### ISSUE TITLE
Order of commits in the git history seems to be broken

### ISSUE THREAD

**Issue by simisimon** [ishepard/pydriller#205]
Hi there! We analyze the commit history of [javascript](https://github.com/airbnb/javascript), but we encountered a problem. It seems that the order of commits in the history is not correct. Using pydriller, we got the following order of commits:
```
1400: 7cef8dad848c5648c6b843e26ea683b9b1f4d43d
1401: 5dec8272e03ce792f3c1a5dc8cf94042609e0615
1402: f5cd2869d35268cc158409195b561b28feca5d94
1403: 5aa203eaa88fb7febfb1ccf8736e896ca0959049
1404: f878edad256fabd4fdcdfdffabd884744b0924d4
1405: 121a95d71524032bfa858271b318aa6e19553dfc
``` 
The problem here is about the commits `1405`, `1404`, and `1400`. Commit `1404` should be the parent of `1405`, but when I checkout commit `1405` and run `git log` my console shows that `1400` is the parent instead. I got the same results when searching for the commits using the GitHub client. Moreover, if I checkout `1404` and use `git log`, the history of `1404` does not lead to `1400`.

At least for me, it seems like a problem regarding the order of commits in the history. However, I'm not sure if this is a real problem or even how it happens. Do you have an explanation for that?

 Thank you in advance for your help.

**Comment by ishepard** [comment]
Hi @simisimon ! 
So internally pydriller runs `git log`. You can try running `git log > tmp.txt` and see the history. 
If I do that, I get:
```
commit 01e39775972e3457defd64af2e94843a848d4fe2
Author: Jordan Harband <ljharb@gmail.com>
Date:   Fri May 5 12:34:00 2017 -0700

commit 121a95d71524032bfa858271b318aa6e19553dfc
Author: Jordan Harband <ljharb@gmail.com>
Date:   Fri May 5 12:16:58 2017 -0700

commit f878edad256fabd4fdcdfdffabd884744b0924d4
Author: Thomas Grainger <tom@procensus.com>
Date:   Fri May 5 11:42:43 2017 +0100


commit 5aa203eaa88fb7febfb1ccf8736e896ca0959049
Author: Jordan Harband <ljharb@gmail.com>
Date:   Fri Sep 1 22:25:31 2017 -0700


commit f5cd2869d35268cc158409195b561b28feca5d94
Author: David Walsh <dlwalsh@gmail.com>
Date:   Tue Jul 4 12:40:53 2017 +1000


commit 5dec8272e03ce792f3c1a5dc8cf94042609e0615
Author: koooge <koooooge@gmail.com>
Date:   Thu Aug 3 17:57:58 2017 +0200


commit 7cef8dad848c5648c6b843e26ea683b9b1f4d43d
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:31:57 2017 -0700

commit ba35e31c76d3c1ab4aeaa7f7967fc4cb98a82787
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:22:41 2017 -0700
```

so the result seems to be correct with what Git says. 

I haven't done an investigation of what's going on though, I just checked that the git history was not broken 😄 
I can help you figure out what's happening if you need it

**Comment by simisimon** [comment]
Well, that is strange. As I mentioned above, when I checkout `121a95d71524032bfa858271b318aa6e19553dfc` and run git log, I get:

```
commit 121a95d71524032bfa858271b318aa6e19553dfc (HEAD)
Author: Jordan Harband <ljharb@gmail.com>
Date:   Fri May 5 12:16:58 2017 -0700

    [eslint config] [deps] [breaking] update `eslint-plugin-jsx-a11y`

commit 7cef8dad848c5648c6b843e26ea683b9b1f4d43d
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:31:57 2017 -0700

    [eslint config] [docs] add rule documentation to `forbid-elements` entry

commit ba35e31c76d3c1ab4aeaa7f7967fc4cb98a82787
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:22:41 2017 -0700

    [eslint config] [breaking] set default React version to 0.15

commit c9c5f7efbf91a7d00c0696f5d4318d6c58f81c5c
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:22:24 2017 -0700

    [eslint config] [minor] enable rules:

     - `jsx-max-props-per-line`
     - `void-dom-elements-no-children`

commit bfc85bddba7b62098798bcbb07d475f2801f3ddf
Author: Jordan Harband <ljharb@gmail.com>
Date:   Mon May 1 12:21:54 2017 -0700

    [eslint config] [deps] update `eslint-config-airbnb-base`, `eslint`
```

If you can help me figure out what is going on, I would be very grateful.

**Comment by simisimon** [comment]
I'm not sure. I was on the master branch,  applied checkout  `<commit_hash>`, and ran git log.  I tried git log with the parameters mentioned in the documentation. But also with these, I got the same history as describe above.

**Comment by ishepard** [comment]
Yeah not sure. The important thing is that Pydriller isn't different from Git. 
So if you find differences, feel free to post it. But to me it looks that in this case they behave the same.

**Comment by simisimon** [comment]
I don't think that this an error, because as you said they behave the same. However, I still don't know why I get different results when using the Github client or when I first checkout the commit and then use git log. It would be interesting to know what the GitHub client does internally when searching for commits.

Anyways, I think I encountered this problem as I expected the git history to be linear, but actually it was not. Maybe that's the reason why I got different parents for a commit.
